test/types/Currency.t.sol

```diff
@@ -4,82 +4,133 @@ pragma solidity ^0.8.20;
+        currencyTest = new CurrencyTest();
+        vm.deal(address(currencyTest), 30 ether);
         nativeCurrency = Currency.wrap(address(0));
         MockERC20 token = new MockERC20("TestA", "A", 18);
-        token.mint(address(this), initialERC20Balance);
+        token.mint(address(currencyTest), initialERC20Balance);
         erc20Currency = Currency.wrap(address(token));
-        erc20Currency.transfer(address(1), sentBalance);
-        nativeCurrency.transfer(address(1), sentBalance);
     }
 
-    function testCurrency_balanceOfSelf_native() public view {
-        assertEq(nativeCurrency.balanceOfSelf(), address(this).balance);
+    function test_fuzz_equals(address a, address b) public pure {
+        assertEq(a == b, Currency.wrap(a) == Currency.wrap(b));
     }
 
-    function testCurrency_balanceOfSelf_token() public view {
-        assertEq(erc20Currency.balanceOfSelf(), initialERC20Balance - sentBalance);
+    function test_fuzz_greaterThan(address a, address b) public pure {
+        assertEq(a > b, Currency.wrap(a) > Currency.wrap(b));
     }
 
-    function testCurrency_balanceOf_native() public view {
-        assertEq(nativeCurrency.balanceOf(otherAddress), sentBalance);
+    function test_fuzz_lessThan(address a, address b) public pure {
+        assertEq(a < b, Currency.wrap(a) < Currency.wrap(b));
     }
 
-    function testCurrency_balanceOf_token() public view {
-        assertEq(erc20Currency.balanceOf(otherAddress), sentBalance);
+    function test_fuzz_greaterThanOrEqualTo(address a, address b) public pure {
+        assertEq(a >= b, Currency.wrap(a) >= Currency.wrap(b));
     }
 
-    function testCurrency_isNative_native_returnsTrue() public view {
-        assertEq(nativeCurrency.isNative(), true);
+    function test_fuzz_balanceOfSelf_native(uint256 amount) public {
+        uint256 balanceBefore = address(currencyTest).balance;
+        amount = bound(amount, 0, balanceBefore);
+        currencyTest.transfer(nativeCurrency, otherAddress, amount);
+        assertEq(balanceBefore - amount, address(currencyTest).balance);
+        assertEq(currencyTest.balanceOfSelf(nativeCurrency), address(currencyTest).balance);
     }
 
-    function testCurrency_isNative_token_returnsFalse() public view {
-        assertEq(erc20Currency.isNative(), false);
+    function test_fuzz_balanceOfSelf_token(uint256 amount) public {
+        amount = bound(amount, 0, initialERC20Balance);
+        currencyTest.transfer(erc20Currency, otherAddress, amount);
+        assertEq(currencyTest.balanceOfSelf(erc20Currency), initialERC20Balance - amount);
+        assertEq(
+            currencyTest.balanceOfSelf(erc20Currency),
+            MockERC20(Currency.unwrap(erc20Currency)).balanceOf(address(currencyTest))
+        );
     }
 
-    function testCurrency_toId_native_returns0() public view {
-        assertEq(nativeCurrency.toId(), uint256(0));
+    function test_fuzz_balanceOf_native(uint256 amount) public {
```

Comment:
can you assert that the currencyTest balance is the prev balance minus amount here too?

Reply:
as well as `test_fuzz_balanceOf_token`, `test_fuzz_transfer_native`, and `test_fuzz_transfer_token`